submissions/Dovahkiin1991/friends-app/app.js

```diff
@@ -0,0 +1,339 @@
+    const value = e.preventDefault();
+    const pipes = [filterByName];
+    
+    if (getOpt('alphabet')) {
+        pipes.push(sortByAlphabet);
+    }
+    if (getOpt('age')) {
+        pipes.push(sortByAge);
+    }
+    if (getOpt('gender')) {
+        pipes.push(filterByGender);
+    }
+
+    renderUsers(pipes);
+});
+
+document.querySelectorAll('.filter__block input[type="radio"]').forEach((elem) => {
+    elem.addEventListener("change", function(event) {
+
+        switch (event.target.name) {
+            case 'alphabet':
+                uncheckRadio('age');
+                renderUsers([
+                    filterByName,
+                    filterByGender,
+                    sortByAlphabet
+                ]);
+
+                break;
+            case 'gender':
+                const pipes = [
+                    filterByName,
+                    filterByGender,
+                ];
+                if (getOpt('alphabet')) {
+                    pipes.push(sortByAlphabet);
+                }
+                if (getOpt('age')) {
+                    pipes.push(sortByAge);
+                }
+                renderUsers(pipes);
+
+                break;
+            case 'age':
+                uncheckRadio('alphabet');
+                renderUsers([
+                    filterByName,
+                    filterByGender,
+                    sortByAge
+                ]);
+                break;
+            default:
+                // renderUsers();
+        }
+    });
+});
+
+const filterByName = (user) => {
+    const searchValue = nameInput.value.trim().toLowerCase();
+    return user.name.first.toLowerCase().includes(searchValue);
+}
+
+const filterByGender = (user) => {
+    const genderRadioOpt = getOpt('gender');
+    if (!genderRadioOpt) { // Pass all if none gender checked
+        return true;
+    }
+
+    if (!['male', 'female'].includes(genderRadioOpt.id)) { // Pass all if 'all' gender checked
+        return true;
+    }
+
+    return genderRadioOpt.id === user.gender;
+}
+
+const sortByAlphabet = (a, b) => {
+    const alphabetRadioOpt = getOpt('alphabet');
+
+    if (alphabetRadioOpt) {
+        let isReversed = alphabetRadioOpt.id === 'alphabetZA';
+        const nameA = a.name.first.toUpperCase();
+        const nameB = b.name.first.toUpperCase();
+        let result;
+
+        // Compare
+        if (nameA < nameB) {
+            result = -1;
+        } else if (nameA > nameB) {
+            result = 1;
+        } else {
+            result = 0;
+        }
+
+        // Reverse if needed
+        if (isReversed) {
+            result *= -1;
+        }
+        return result;
+    }
+
+    return 0;
+}
+
+const sortByAge = (userA, userB) => {
+    const ageRadioOpt = getOpt('age');
+
+    if (ageRadioOpt) {
+        let isReversed = ageRadioOpt.id === 'ageHighest';
+        return isReversed
+            ? userB.dob.age - userA.dob.age
+            : userA.dob.age - userB.dob.age;
+    }
+
+    return 0;
+}
```

Comment:
First of all, the same comments as to filtering functions.
Secondly, sorting functions essentially work in the same way. They check a and b with some operations and return -1, 1, or 0. Find a way to produce a universal sorting function. It will drastically shorten your code and make it DRY. The trick is then just to pass correct arguments to such a function.